tests/shutdown/grub_set_bootargs.pm

```diff
@@ -16,16 +16,32 @@ use testapi;
 
 sub run() {
     select_console('root-console');
-    script_run "source /etc/default/grub";
-    script_run 'new_cmdline=`echo $GRUB_CMDLINE_LINUX_DEFAULT | sed \'s/\(^\| \)quiet\($\| \)/ /\'`';
-    if (get_var("ENABLE_FIPS")) {
-        script_run 'new_cmdline="$new_cmdline fips=1"';
-        if (get_var("ENCRYPT") && !get_var("FULL_LVM_ENCRYPT")) {
-            script_run 'new_cmdline="$new_cmdline boot=$(df /boot | tail -1 | cut -d" " -f1)"';
+    my @cmds;
+    push @cmds, "source /etc/default/grub";
+    push @cmds, 'new_cmdline=`echo $GRUB_CMDLINE_LINUX_DEFAULT | sed \'s/\(^\| \)quiet\($\| \)/ /\'`';
```

Comment:
after this line you can just add
 
```
 if (get_var("ENABLE_FIPS")) {
             push @cmds,  'new_cmdline="$new_cmdline fips=1"';
             if (get_var("ENCRYPT") && !get_var("FULL_LVM_ENCRYPT")) {
                 push @cmds,  'new_cmdline="$new_cmdline boot=$(df /boot | tail -1 | cut -d" " -f1)"';
             }
        }
```

this will allow to replace all complex logic in 26-42 lines with 
```
for my $cmd (@cmds) {
            if (check_var('ARCH', 'aarch64') && check_var('VERSION', '12-SP2')) { 
             type_string $cmd . "\n", wait_screen_change => 1;
            }else {
              script_run $cmd;
             }
         }
```